### ISSUE TITLE
REST-API returns confusing 404 'invalid endpoint' error when called w/ non-checksum addresses

### ISSUE THREAD

**Issue by ulope** [raiden-network/raiden#1667]
## Problem Definition

Since #1506 was merged the REST-API (amongst other places) no longer accepts non-checksum (EIP55) addresses. 

Unfortunately, the error message returned if a non-checksum address is given to an endpoint is a rather confusing `{"errors": "invalid endpoint"}` with a HTTP status `404`.

We should return a proper error message informing the user about using checksummed addresses.

**Comment by palango** [comment]
We found out about the same problem today when we tested the web ui.On 26 Jun 2018 7:30 PM, Ulrich Petri <notifications@github.com> wrote:Problem Definition
Since #1506 was merged the REST-API (amongst other places) no longer accepts non-checksum (EIP55) addresses.
Unfortunately, the error message returned if a non-checksum address is given to an endpoint is a rather confusing {"errors": "invalid endpoint"} with a HTTP status 404.
We should return a proper error message informing the user about using checksummed addresses.
